src/woo_search/search_index/api/serializers/publications.py

```diff
@@ -51,12 +51,26 @@ class DocumentSerializer(serializers.Serializer):
         )
     )
     identifier = serializers.CharField(
-        help_text=_("The (primary) unique identifier."),
+        help_text=_("**DEPRECATED** The (primary) unique identifier."),
         max_length=255,
         required=False,
         allow_blank=True,
         default="",
     )
+    identifiers = serializers.ListField(
+        help_text=_("The document identifiers attached to this document."),
+        child=serializers.CharField(
+            max_length=40,
+            help_text=_(
+                "An identifier specific to this document. Note that multiple "
+                "documents can share identifiers, as additional context is "
+                "required to uniquely identify it, but this context "
+                "is deliberately not indexed."
+            ),
+        ),
+        required=False,
+        default=[],
```

Comment:
don't use mutable defaults!

```suggestion
        default=list,
```